GitUI/CommandsDialogs/FormCleanupRepository.cs

```diff
@@ -24,34 +24,45 @@ public FormCleanupRepository(GitUICommands commands)
             InitializeComponent();
             InitializeComplete();
             PreviewOutput.ReadOnly = true;
+
             checkBoxPathFilter_CheckedChanged(this, EventArgs.Empty);
+            checkBoxExcludePathFilter_CheckedChanged(this, EventArgs.Empty);
         }
 
         public void SetPathArgument(string? path)
         {
             if (string.IsNullOrEmpty(path))
             {
-                checkBoxPathFilter.Checked = false;
-                textBoxPaths.Text = "";
+                checkBoxExcludePathFilter.Checked = false;
+                textBoxExcludePaths.Text = "";
+
+                checkBoxIncludePathFilter.Checked = false;
+                textBoxIncludePaths.Text = "";
             }
             else
             {
-                checkBoxPathFilter.Checked = true;
-                textBoxPaths.Text = path;
+                checkBoxExcludePathFilter.Checked = true;
+                textBoxExcludePaths.Text = path;
+
+                checkBoxIncludePathFilter.Checked = true;
+                textBoxIncludePaths.Text = path;
             }
         }
 
         private void Preview_Click(object sender, EventArgs e)
         {
-            string pathArgument = GetPathArgumentFromGui();
+            string includePathArgument = GetInclusivePathArgumentFromGui();
+            string excludePathArguments = GetExclusivePathArgumentFromGui();
             CleanMode mode = GetCleanMode();
-            var cleanUpCmd = GitCommandHelpers.CleanCmd(mode, dryRun: true, directories: RemoveDirectories.Checked, paths: pathArgument);
+            var cleanUpCmd = GitCommandHelpers.CleanCmd(mode, dryRun: false, directories: RemoveDirectories.Checked,
+                paths: includePathArgument, excludes: excludePathArguments);
+
             string cmdOutput = FormProcess.ReadDialog(this, arguments: cleanUpCmd, Module.WorkingDir, input: null, useDialogSettings: true);
             PreviewOutput.Text = EnvUtils.ReplaceLinuxNewLinesDependingOnPlatform(cmdOutput);
 
             if (CleanSubmodules.Checked)
             {
-                var cleanSubmodulesCmd = GitCommandHelpers.CleanSubmodules(mode, dryRun: true, directories: RemoveDirectories.Checked, paths: pathArgument);
+                var cleanSubmodulesCmd = GitCommandHelpers.CleanSubmodules(mode, dryRun: true, directories: RemoveDirectories.Checked, paths: includePathArgument);
```

Comment:
❔ Should we apply `excludePathArguments` for submodules, too?

Reply:
I see two viewpoints:
* Paths are relative the superproject and only should be relative the master. Hard to implement though, if wildcards are to be allowed.
* apply the same as for the superproject, similar to include paths. Easier to implement, maybe not expected?

Probably second option.
@scottsumrall thoughts?

Reply:
I'm not sure I completely follow. As of right now, the excludes for the general clean are only for excluding individual files and do not support wildcards. I assume it would be the same for submodules? If we want this feature included I would lean towards the first solution but the second option seems fine for its current state.

Reply:
The first option would affect the include files too. But that wasrecently added without consideration.
However, I do not find it reasonable.
Use second alternative, in this PR before merge.

Add exclude paths to `CleanSubmodules()`
(Make sure there is a test as well, also for include not tested at all now)